lib/Fetcher/FeedFetcher.php

```diff
@@ -338,15 +340,18 @@ protected function getFavicon(FeedInterface $feed, string $url)
 
         $is_image = substr(mime_content_type($favicon_path), 0, 5) === "image";
 
+        $base_url = new Net_URL2($url);
+        $base_url->setPath('');
+
         // check if file is actually an image
         if (!$is_image) {
-            return $this->faviconFactory->get($url);
+            return $this->faviconFactory->get($base_url->getNormalizedURL());
         }
 
         list($width, $height, $type, $attr) = getimagesize($favicon_path);
         // check if image is square else fall back to favicon
         if ($width !== $height) {
-            return $this->faviconFactory->get($url);
+            return $this->faviconFactory->get($base_url);
```

Comment:
Is this correct? Or should it be `$base_url->getNormalizedURL()` (as at line 348)?

Reply:
Well, I learned that the favicon lib does already do something like that so it's probably pointless to try.
But yes it should have been `$base_url->getNormalizedURL()` just to be safe.